src/main/java/org/threadly/concurrent/CentralThreadlyPool.java

```diff
@@ -0,0 +1,215 @@
+   * this pool from possibly fully consuming the central pool.  If {@code maxThreads <= 0} then 
+   * tasks submitted to this pool will be able to use all available threads as long as they don't 
+   * end up consuming threads which other pools might need reserved.
+   * 
+   * @param minThreads Minimum number of threads the provided pool should be guaranteed to have, negative is ignored
+   * @param maxThreads Maximum number of threads to the returned pool can consume, or negative to use any available
+   * @return A pool with the requested specifications for task scheduling or execution
+   */
+  public static SchedulerService rangedThreadCountPool(int minThreads, int maxThreads) {
+    if (maxThreads == 1) {
+      return singleThreadPool(minThreads > 0);
+    } else {
+      return new GuaranteedConnectionProtectingScheduler(MASTER_SCHEDULER.getDefaultPriority(), 
+                                                         minThreads, maxThreads);
+    }
+  }
+  
+  private static PrioritySchedulerService masterScheduler(TaskPriority defaultPriority) {
+    if (defaultPriority == TaskPriority.High) {
+      return MASTER_SCHEDULER;
+    } else {  // starvable is assumed to not be used in this current implementation
+      return LOW_PRIORITY_MASTER_SCHEDULER;
+    }
+  }
+  
+  protected static class PoolResizer {
+    private final int amount;
+    
+    public PoolResizer(int amount) {
+      this.amount = amount;
+
+      MASTER_SCHEDULER.adjustPoolSize(amount);
+      GUARANTEED_THREADS.add(amount);
+    }
+    
+    @Override
+    protected void finalize() {
+      GUARANTEED_THREADS.add(-amount);
+      MASTER_SCHEDULER.adjustPoolSize(-amount);
+    }
+  }
+
+  // TODO - this is slightly flawed, we need to have a way for the pool to grow without 
+  //          requiring protected threads otherwise we will never use available resources.
+  //          If it does not make sense changing when the pool is expanded (or how), then this 
+  //          class is probably providing a protection that makes everything useless...quite a thinker
+  protected static class GuaranteedConnectionProtectingScheduler extends SchedulerServiceLimiter {
+    @SuppressWarnings("unused")
+    private final Object gcReference; // object just held on to track garbage collection
+    private final int personallyGarunteedThreads;
+    private final int maxThreads;
+    
+    public GuaranteedConnectionProtectingScheduler(TaskPriority priority, 
+                                                  int personallyGarunteedThreads, int maxThreads) {
+      super(masterScheduler(priority), 1);
+      
+      this.gcReference = personallyGarunteedThreads > 0 ? new PoolResizer(personallyGarunteedThreads) : null;
+      this.personallyGarunteedThreads = personallyGarunteedThreads > 0 ? personallyGarunteedThreads : 0;
+      this.maxThreads = maxThreads < 1 ? Integer.MAX_VALUE : maxThreads;
```

Comment:
I feel like saying I need "X" max threads should make the thread pool at least that size, and if they say N dedicated the pool should be at least X with N not being shared on another pool.  Basically whatever the most threads requested for a pool is is how many threads should be in the global pool at a minimum, plus any dedicated ones or any other pools

so in a case of these sub pool requests:
max    dedicated
10         5
5           1
20         3
50         1

there would be 60 threads, 50 for general processing and 10 dedicated.  if the last pool GC's we would adjust to have 29 threads total, 20 shared and 9 dedicated.